### PR TITLE
Added implementation files for C Object Detection APIs

### PR DESCRIPTION
**Pull request by priankakariatyml**

### CODE REVIEW

File: tensorflow_lite_support/c/task/vision/object_detector.cc [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  StatusOr<std::unique_ptr<FrameBufferCpp>> cpp_frame_buffer_status =
+      ::tflite::task::vision::CreateCppFrameBuffer(frame_buffer);
+  if (!cpp_frame_buffer_status.ok()) {
+    tflite::support::CreateTfLiteSupportErrorWithStatus(
+        cpp_frame_buffer_status.status(), error);
+    return nullptr;
+  }
+
+  StatusOr<DetectionResultCpp> cpp_detection_result_status =
+      detector->impl->Detect(*std::move(cpp_frame_buffer_status.value()));
```
Comment (lu-wang-g) [inline tensorflow_lite_support/c/task/vision/object_detector.cc:187]: Is `std::move` needed here?

File: tensorflow_lite_support/c/task/vision/object_detector.cc [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+#endif  // __cplusplus
+
+struct TfLiteObjectDetector {
+  std::unique_ptr<ObjectDetectorCpp> impl;
+};
+
+TfLiteObjectDetectorOptions TfLiteObjectDetectorOptionsCreate() {
+  // Use brace-enclosed initializer list will break the Kokoro test.
+  TfLiteObjectDetectorOptions options = {{{0}}};
+  options.classification_options.max_results = -1;
```
[the remark this answers is not in the document]
Comment (priankakariatyml) [reply tensorflow_lite_support/c/task/vision/object_detector.cc]: Fixed

File: tensorflow_lite_support/c/task/vision/object_detector.cc [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  options.classification_options.score_threshold = 0.0;
```
[the remark this answers is not in the document]
Comment (priankakariatyml) [reply tensorflow_lite_support/c/task/vision/object_detector.cc]: I have changed it to -FLT_MAX

File: tensorflow_lite_support/c/task/processor/detection_result.cc [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+extern "C" {
+#endif  // __cplusplus
+
+void TfLiteDetectionResultDelete(TfLiteDetectionResult* detection_result) {
+  for (int i = 0; i < detection_result->size; ++i) {
+    TfLiteDetection detections = detection_result->detections[i];
+    for (int j = 0; j < detections.size; ++j) {
+      // `strdup` obtains memory using `malloc` and the memory needs to be
+      // released using `free`.
+      free(detections.categories[j].display_name);
```
[the remark this answers is not in the document]
Comment (priankakariatyml) [reply tensorflow_lite_support/c/task/processor/detection_result.cc]: Fixed

File: tensorflow_lite_support/c/task/vision/object_detector.cc (hunk 1 above)
Comment (priankakariatyml) [reply tensorflow_lite_support/c/task/vision/object_detector.cc:188]: Not needed. Fixed in both image_classifier and object_detector

File: tensorflow_lite_support/c/task/core/utils/base_options_utils.cc [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+namespace task {
+
+TfLiteBaseOptions GetDefaultBaseOptions() {
+  TfLiteBaseOptions base_options = {0};
+  base_options.compute_settings.cpu_settings.num_threads = -1;
+  return base_options;
+}
+
+}  // namespace task
+}  // namespace tflite
```
Comment (lu-wang-g) [inline tensorflow_lite_support/c/task/core/utils/base_options_utils.cc:28]: nit: add an end-of-file blank line. Same for the other files (where you see the red sign of a circle with a `-`.).

File: tensorflow_lite_support/c/task/core/utils/base_options_utils.cc [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/c/task/core/utils/base_options_utils.h"
+
+namespace tflite {
+namespace task {
+
+TfLiteBaseOptions GetDefaultBaseOptions() {
```
Comment (lu-wang-g) [inline tensorflow_lite_support/c/task/core/utils/base_options_utils.cc:21]: nit: GetDefaultBaseOptions => CreateDefaultBaseOptions

File: tensorflow_lite_support/c/task/processor/utils/classification_options_utils.cc [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+==============================================================================*/
+
+#include "tensorflow_lite_support/c/task/processor/utils/classification_options_utils.h"
+
+#include <float.h>
+
+namespace tflite {
+namespace task {
+
+TfLiteClassificationOptions GetDefaultClassificationOptions() {
```
Comment (lu-wang-g) [inline tensorflow_lite_support/c/task/processor/utils/classification_options_utils.cc:23]: nit: GetDefaultClassificationOptions => CreateDefaultClassificationOptions

File: tensorflow_lite_support/c/task/processor/utils/classification_options_utils.cc [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/c/task/processor/utils/classification_options_utils.h"
+
+#include <float.h>
+
+namespace tflite {
+namespace task {
```
Comment (lu-wang-g) [inline tensorflow_lite_support/c/task/processor/utils/classification_options_utils.cc:21]: Within `namespace processor`?

**Review by lu-wang-g — COMMENTED** [review]
Looks great! Are we going to add tests in a follow up PR?

File: tensorflow_lite_support/c/task/processor/utils/classification_options_utils.cc (hunk 8 above)
Comment (priankakariatyml) [reply tensorflow_lite_support/c/task/processor/utils/classification_options_utils.cc:21]: Fixed in processor and coree

**Comment by priankakariatyml** [comment]
> Looks great! Are we going to add tests in a follow up PR?

I think a follow up PR is better. Or if you want me to add it in this one, I can.

**Review by lu-wang-g — APPROVED** [review]

**Pull request merged**